DOCS-1691: Docs for ML Models in registry (#2447)
viamrobotics/docs: docs/ml/upload-model.md

  - /manage/data/upload-model/
  - /manage/ml/upload-model/
description: "Upload a Machine Learning model to Viam to use it with the ML Model service."
# SME: Steven B.
---

Upload a new model to your organization with the option to share it in the [Viam registry](https://app.viam.com/registry).

> Or, upload a new version of a published model.

Author: Ok cool! I don't think it's quite the same steps though, so adjusting that wording a bit

Select **Next steps** to continue.

{{<imgproc src="/ml/upload-model.png" resize="600x" alt="Upload model menu on the DATA tab of the Viam app.">}}

1. Specify a name for the model.
2. Specify the **Model type**.
3. Add a `.tflite` model file.
4. Add a `.txt` label file.
5. Add a short description.
   This will help other users to understand how to use your model.

Select **Upload model**.
The model is now visible on the **MODELS** subtab of the **DATA** tab in the [Viam app](https://app.viam.com) and available to [deploy on your machine](/ml/deploy/).
Also, if you selected **Public**, it should be publicly visible in the [Viam registry](https://app.viam.com/registry).

{{% /tab %}}
{{% tab name="New version" %}}

{{<imgproc src="/ml/select-existing-model.png" resize="400x" alt="Select from your existing models.">}}

1. Select the model you would like to update from your existing models.
2. Select **Next steps** to continue.
3. Add an updated `.tflite` model file.
4. Add an updated `.txt` label file.
5. Click **Upload model**.

Your model is now updated.

> {{% alert title="Note" color="note" %}}

Reviewer: I think this note may need checking. I think I saw yesterday that the **COPY** button is no longer there and it's now a slightly different UI? Can you double check?

Author: You are correct about that, updated!

  - /manage/data/upload-model/
  - /manage/ml/upload-model/
description: "Upload a Machine Learning model to Viam to use it with the ML Model service."
# SME: Steven B.
---

The ML model service works with models trained inside and outside the Viam app.
To use a model that you have trained yourself outside the Viam app, [upload it as a new model privately or share it in the Viam registry](#upload-a-new-model-or-new-version).
If you need to update a previously uploaded model, you can also [upload a new version](#upload-a-new-model-or-new-version).

> If you have [trained](/ml/train-model/) or uploaded an ML model privately and now want to make it available for reuse, you can at any point [make the existing model public in the registry](/#make-an-existing-model-public-in-the-registry).

Author: ```suggestion
If you have [trained](/ml/train-model/) or uploaded an ML model privately and now want to make it available for reuse, you can at any point [make the existing model public in the registry](#make-an-existing-model-public-in-the-registry).
```

  - /manage/data/upload-model/
  - /manage/ml/upload-model/
description: "Upload a Machine Learning model to Viam to use it with the ML Model service."
# SME: Steven B.
---

> ## Upload an existing model to the registry

Reviewer: This page needs an introduction for the two parts to this? Users may land here and not have uploaded their model to Viam yet. The second part of this page should almost certainly come first because it's the more general case. This one is really more "if you later want to make a model public you can". And the title there shouldn't be uploading (it's already uploaded), it should be `Make and exisitng model public in the registry` or something like that

> Trained ML models are available in the [Viam registry](https://app.viam.com/registry) for public access.

Reviewer: Only if you make them accessible! This leaves room for confusion.

Author: Removed

To add a model that you've trained to the [Viam registry](https://app.viam.com/registry) so that other users can deploy it onto their robots:

1. On the [**DATA** tab](https://app.viam.com/data/view) in the Viam app, navigate to the **MODELS** subtab.

>    A list of all of your ML models is displayed.

Reviewer: Avoid passive voice
```suggestion
   You'll see a list of all ML models you have access to.
```

2. Open the menu on the right side of an ML model's row.

{{<imgproc src="/ml/model-list.png" resize="1000x" alt="List of models displayed on MODELS subtab of DATA tab.">}}

3. Then, click **Make public in Registry**.

{{<imgproc src="/ml/publish-model.png" resize="600x" alt="Publish model to registry action card.">}}

4. Add a short description.
   This will help other users to understand how to use your model.
5. Then, click **Publish model**.

>    Your model should be publicly visible in the [Viam registry](https://app.viam.com/registry).

Reviewer: Avoid `should` - it sounds like you're not sure the product works/will work
```suggestion
   Your model is now publicly visible in the [Viam registry](https://app.viam.com/registry).
```

Select **Private** to only publish the model for internal use within this {{< glossary_tooltip term_id="organization" text="organization" >}}.
Select **Next steps** to continue.

{{<imgproc src="/ml/upload-model.png" resize="600x" alt="Upload model menu on the DATA tab of the Viam app.">}}

1. Specify a name for the model.
2. Specify the **Model type**.
3. Add a `.tflite` model file.
4. Add a `.txt` label file.
5. Add a short description.
   This will help other users to understand how to use your model.

Select **Upload model**.
The model should now be visible on the **MODELS** subtab of the **DATA** tab in the [Viam app](https://app.viam.com) and available to [deploy on your machine](/ml/deploy/).
Also, if you selected **Public**, it should be publicly visible in the [Viam registry](https://app.viam.com/registry).

{{% /tab %}}
{{% tab name="New version" %}}

{{<imgproc src="/ml/select-existing-model.png" resize="400x" alt="Select from your existing models.">}}

1. Select the model you would like to update from your existing models.
2. Select **Next steps** to continue.
3. Add an updated `.tflite` model file.
4. Add an updated `.txt` label file.
5. Click **Upload model**.

> Your model is now replaced with the updated version.

Author: Didn't realize that was still applicable, apologies. Now that I look closely it seems like the two parts are actually different things.
